pkg/generators/forwarding/fluentd/templates.go

```diff
@@ -598,6 +597,21 @@ const pipelineToOutputCopyTemplate = `{{- define "pipelineToOutputCopyTemplate"
 
 const outputLabelConfTemplate = `{{- define "outputLabelConf" -}}
 <label {{.LabelName}}>
+  {{ if (.NeedChangeElasticsearchIndexName) -}}
+  <filter **>
+    @type record_modifier
+	<record>
+	  indexFromKey     ${record['kubernetes']['labels']['{{.Target.OutputTypeSpec.Elasticsearch.IndexKey}}']}
```

Comment:
This is incorrect: we want to treat IndexKey as a full key in the record. As perr the PR, it could also be kubernetes.namespace_name or kuberntes.openshift.label.XX

/lgtm cancel

Reply:
@alanconway PTAL. tried to fix it.